ios/brave-ios/Sources/Brave/Frontend/Rewards/BraveRewards.swift

```diff
@@ -220,25 +220,27 @@ public class BraveRewards: NSObject {
   func reportLoadedPage(
     redirectChain: [URL],
     tabId: Int,
-    html: String,
+    html: String?,
     adsInnerText: String?
   ) {
     guard let url = redirectChain.last else {
       return
     }
 
     tabRetrieved(tabId, url: url, html: html)
-    if let innerText = adsInnerText, ads.isServiceRunning() {
+    if let html = html, ads.isServiceRunning() {
       ads.notifyTabHtmlContentDidChange(
         tabId,
-        redirectChain: redirectChain.isEmpty ? [url] : redirectChain,
+        redirectChain: redirectChain,
         html: html
       )
-      ads.notifyTabTextContentDidChange(
-        tabId,
-        redirectChain: redirectChain.isEmpty ? [url] : redirectChain,
-        text: innerText
-      )
+      if let innerText = adsInnerText {
+        ads.notifyTabTextContentDidChange(
```

Comment:
This will only run if html is `non-nil`, is that the expected behaviour or should it run as long as `adsInnerText` is non-nil?

Reply:
@aseren html and text should be separated, as mentioned by @kylehickinson , thanks

Reply:
The previous implementation (code in the current `master` branch) reported HTML/text only if `adsInnerText` is non-nil. Now this is fixed,  so HTML and Text are separated.